src/cloudai/cli/handlers.py

```diff
@@ -195,34 +196,47 @@ def register_signal_handlers(signal_handler: Callable) -> None:
         signal.signal(sig, signal_handler)
 
 
-def handle_dry_run_and_run(args: argparse.Namespace) -> int:
+def _setup_system_and_scenario(
+    args: argparse.Namespace,
+) -> tuple[Optional[System], Optional[TestScenario], Optional[list[Test]]]:
     parser = Parser(args.system_config)
-    system, tests, test_scenario = parser.parse(args.tests_dir, args.test_scenario)
+    try:
```

Comment:
Can we parse each config type separately and handle relevant exceptions? Looks like `.parse` is too high level, we can't do efficient error handling now.